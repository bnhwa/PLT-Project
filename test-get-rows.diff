1c1
< 3
---
> 3.00
\ No newline at end of file
