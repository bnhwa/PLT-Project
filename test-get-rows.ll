; ModuleID = 'xirtam'
source_filename = "xirtam"

%struct.matrix = type { i32, i32, double*, i32 }

@fmt = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1

declare i32 @printf(i8*, ...)

declare i32 @display(%struct.matrix*)

declare %struct.matrix* @initMatrix_CG(i32, i32)

declare %struct.matrix* @storeVal(%struct.matrix*, double)

declare double @pub_get(%struct.matrix*, double, double)

declare i32 @pub_set(%struct.matrix*, double, double, double)

declare %struct.matrix* @matrixMult(%struct.matrix*, %struct.matrix*)

declare %struct.matrix* @mAdd(%struct.matrix*, %struct.matrix*)

declare double @mNumRows(%struct.matrix*)

define i32 @main() {
entry:
  %r = alloca double
  %m = alloca %struct.matrix*
  %matrix_init = call %struct.matrix* @initMatrix_CG(i32 3, i32 3)
  %store_val = call %struct.matrix* @storeVal(%struct.matrix* %matrix_init, double -1.000000e+00)
  %store_val1 = call %struct.matrix* @storeVal(%struct.matrix* %matrix_init, double 1.000000e+00)
  %store_val2 = call %struct.matrix* @storeVal(%struct.matrix* %matrix_init, double 1.000000e+00)
  %store_val3 = call %struct.matrix* @storeVal(%struct.matrix* %matrix_init, double 1.000000e+00)
  %store_val4 = call %struct.matrix* @storeVal(%struct.matrix* %matrix_init, double 2.000000e+00)
  %store_val5 = call %struct.matrix* @storeVal(%struct.matrix* %matrix_init, double 4.000000e+00)
  %store_val6 = call %struct.matrix* @storeVal(%struct.matrix* %matrix_init, double 1.000000e+00)
  %store_val7 = call %struct.matrix* @storeVal(%struct.matrix* %matrix_init, double 2.000000e+00)
  %store_val8 = call %struct.matrix* @storeVal(%struct.matrix* %matrix_init, double 3.000000e+00)
  store %struct.matrix* %matrix_init, %struct.matrix** %m
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), double 3.000000e+00)
  ret i32 0
}
