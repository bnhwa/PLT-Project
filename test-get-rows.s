	.text
	.file	"xirtam"
	.section	.rodata.cst8,"aM",@progbits,8
	.p2align	3               # -- Begin function main
.LCPI0_0:
	.quad	-4616189618054758400    # double -1
.LCPI0_1:
	.quad	4607182418800017408     # double 1
.LCPI0_2:
	.quad	4611686018427387904     # double 2
.LCPI0_3:
	.quad	4616189618054758400     # double 4
.LCPI0_4:
	.quad	4613937818241073152     # double 3
	.text
	.globl	main
	.p2align	4, 0x90
	.type	main,@function
main:                                   # @main
	.cfi_startproc
# %bb.0:                                # %entry
	pushq	%rbx
	.cfi_def_cfa_offset 16
	subq	$16, %rsp
	.cfi_def_cfa_offset 32
	.cfi_offset %rbx, -16
	movl	$3, %edi
	movl	$3, %esi
	callq	initMatrix_CG@PLT
	movq	%rax, %rbx
	movsd	.LCPI0_0(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%rax, %rdi
	callq	storeVal@PLT
	movsd	.LCPI0_1(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%rbx, %rdi
	callq	storeVal@PLT
	movq	%rbx, %rdi
	movsd	.LCPI0_1(%rip), %xmm0   # xmm0 = mem[0],zero
	callq	storeVal@PLT
	movq	%rbx, %rdi
	movsd	.LCPI0_1(%rip), %xmm0   # xmm0 = mem[0],zero
	callq	storeVal@PLT
	movsd	.LCPI0_2(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%rbx, %rdi
	callq	storeVal@PLT
	movsd	.LCPI0_3(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%rbx, %rdi
	callq	storeVal@PLT
	movq	%rbx, %rdi
	movsd	.LCPI0_1(%rip), %xmm0   # xmm0 = mem[0],zero
	callq	storeVal@PLT
	movq	%rbx, %rdi
	movsd	.LCPI0_2(%rip), %xmm0   # xmm0 = mem[0],zero
	callq	storeVal@PLT
	movsd	.LCPI0_4(%rip), %xmm0   # xmm0 = mem[0],zero
	movq	%rbx, %rdi
	callq	storeVal@PLT
	movq	%rbx, (%rsp)
	leaq	.Lfmt(%rip), %rdi
	movsd	.LCPI0_4(%rip), %xmm0   # xmm0 = mem[0],zero
	movb	$1, %al
	callq	printf@PLT
	xorl	%eax, %eax
	addq	$16, %rsp
	.cfi_def_cfa_offset 16
	popq	%rbx
	.cfi_def_cfa_offset 8
	retq
.Lfunc_end0:
	.size	main, .Lfunc_end0-main
	.cfi_endproc
                                        # -- End function
	.type	.Lfmt,@object           # @fmt
	.section	.rodata.str1.1,"aMS",@progbits,1
.Lfmt:
	.asciz	"%g\n"
	.size	.Lfmt, 4

	.section	".note.GNU-stack","",@progbits
