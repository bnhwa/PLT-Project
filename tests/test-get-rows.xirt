num main() {
	printn(1000);
    xirtam thematrix;
    num rows;
    thematrix = [[1, 2, 3], [1, 2, 3], [1, 2, 3]];
    rows = matnumrows(thematrix);
    printn(rows);
}