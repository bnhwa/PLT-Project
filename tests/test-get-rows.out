1000
3.000000